File: PSW-projekat/hospital-be-main/src/HospitalLibrary/Core/DTOs/SchedulingStatisticsDTO.cs
using System.Collections.Generic;

namespace HospitalLibrary.Core.DTOs
{
    public class SchedulingStatisticsDTO
    {
        public double AvarageSchedulingDuration { get; set; }
        public int LinearSchedulingNumber { get; set; }
        public int NonlinearSchedulingNumber { get; set; }
        public List<int> FinishedSchedulingsPerDay { get; set; }
        public List<int> UnfinishedSchedulingsPerDay { get; set; }
        public double AvarageNumberOfStepsForSuccessfulScheduling { get; set; }
        public List<double> AvarageNumberOfEachStepForSuccessfulScheduling { get; set; }
        public List<NumberOfFinishedAndUnfinishedSchedulingForPatient> NumberOfFinishedAndUnfinishedSchedulingForAllPatients { get; set; }


        public SchedulingStatisticsDTO
        (
            double avarageSchedulingDuration,
            int linearSchedulingNumber,
            int nonlinearSchedulingNumber,
            List<int> finishedSchedulingsPerDay,
            List<int> unfinishedSchedulingsPerDay,
            double avarageNumberOfStepForSuccessfulScheduling,
            List<double> avarageNumberOfEachStepForSuccessfulScheduling,
            List<NumberOfFinishedAndUnfinishedSchedulingForPatient> numberOfFinishedAndUnfinishedSchedulingForAllPatients
        )
        {
            AvarageSchedulingDuration = avarageSchedulingDuration;
            LinearSchedulingNumber = linearSchedulingNumber;
            NonlinearSchedulingNumber = nonlinearSchedulingNumber;
            FinishedSchedulingsPerDay = finishedSchedulingsPerDay;
            UnfinishedSchedulingsPerDay = unfinishedSchedulingsPerDay;
            AvarageNumberOfStepsForSuccessfulScheduling = avarageNumberOfStepForSuccessfulScheduling;
            AvarageNumberOfEachStepForSuccessfulScheduling = avarageNumberOfEachStepForSuccessfulScheduling;
            NumberOfFinishedAndUnfinishedSchedulingForAllPatients = numberOfFinishedAndUnfinishedSchedulingForAllPatients;
        } 
    }
}

File: PSW-projekat/PZL-Angular/src/app/modules/hospital/model/scheduling-statistics-dto.ts
export class SchedulingStatisticsDTO {
    avarageSchedulingDuration : number;
    linearSchedulingNumber : number;
    nonlinearSchedulingNumber : number;
    finishedSchedulingsPerDay : number[];
    unfinishedSchedulingsPerDay : number[];
    avarageNumberOfStepsForSuccessfulScheduling : number;
    avarageNumberOfEachStepForSuccessfulScheduling : number[];
    numberOfFinishedAndUnfinishedSchedulingForAllPatients : Array<NumberOfFinishedAndUnfinishedSchedulingForPatient>;


    public constructor(avarageSchedulingDuration: number, linearSchedulingNumber: number, nonlinearSchedulingNumber: number,
                        finishedSchedulingsPerDay: number[], unfinishedSchedulingsPerDay: number[],
                        avarageNumberOfStepForSuccessfulScheduling: number,
                        avarageNumberOfEachStepForSuccessfulScheduling: number[], 
                        numberOfFinishedAndUnfinishedSchedulingForAllPatients: Array<NumberOfFinishedAndUnfinishedSchedulingForPatient>) {
        {
            this.avarageSchedulingDuration = avarageSchedulingDuration;
            this.linearSchedulingNumber = linearSchedulingNumber;
            this.nonlinearSchedulingNumber = nonlinearSchedulingNumber;
            this.finishedSchedulingsPerDay = finishedSchedulingsPerDay;
            this.unfinishedSchedulingsPerDay = unfinishedSchedulingsPerDay;
            this.avarageNumberOfStepsForSuccessfulScheduling = avarageNumberOfStepForSuccessfulScheduling;
            this.avarageNumberOfEachStepForSuccessfulScheduling = avarageNumberOfEachStepForSuccessfulScheduling;
            this.numberOfFinishedAndUnfinishedSchedulingForAllPatients = numberOfFinishedAndUnfinishedSchedulingForAllPatients;
        }
    }
}

export class NumberOfFinishedAndUnfinishedSchedulingForPatient {
    patientId : number;
    fullName : string;
    numberOfFinishedSchedulings : number;
    numberOfUnfinishedSchedulings : number;

    public constructor(patientId : number, fullName : string, numberOfFinishedSchedulings : number, numberOfUnfinishedSchedulings : number) {
        this.patientId = patientId;
        this.fullName = fullName;
        this.numberOfFinishedSchedulings = numberOfFinishedSchedulings;
        this.numberOfUnfinishedSchedulings = numberOfUnfinishedSchedulings;
    }
}